SimCalorimetry/EcalSimAlgos/src/EEShape.cc

```diff
@@ -2,282 +2,276 @@
-   aVec[142] = 0.968989;
-   aVec[143] = 0.972846;
-   aVec[144] = 0.977824;
-   aVec[145] = 0.981341;
-   aVec[146] = 0.982743;
-   aVec[147] = 0.984082;
-   aVec[148] = 0.986609;
-   aVec[149] = 0.985925;
-   aVec[150] = 0.984707;
-   aVec[151] = 0.981384;
-   aVec[152] = 0.980941;
-   aVec[153] = 0.980132;
-   aVec[154] = 0.978176;
-   aVec[155] = 0.975987;
-   aVec[156] = 0.973396;
-   aVec[157] = 0.970395;
-   aVec[158] = 0.967422;
-   aVec[159] = 0.96346;
-   aVec[160] = 0.959622;
-   aVec[161] = 0.955396;
-   aVec[162] = 0.950846;
-   aVec[163] = 0.945995;
-   aVec[164] = 0.940835;
-   aVec[165] = 0.935373;
-   aVec[166] = 0.929544;
-   aVec[167] = 0.923836;
-   aVec[168] = 0.91787;
-   aVec[169] = 0.910812;
-   aVec[170] = 0.903924;
-   aVec[171] = 0.897669;
-   aVec[172] = 0.89127;
-   aVec[173] = 0.884613;
-   aVec[174] = 0.878051;
-   aVec[175] = 0.867734;
-   aVec[176] = 0.856756;
-   aVec[177] = 0.848477;
-   aVec[178] = 0.842189;
-   aVec[179] = 0.833787;
-   aVec[180] = 0.825903;
-   aVec[181] = 0.817931;
-   aVec[182] = 0.8102;
-   aVec[183] = 0.802771;
-   aVec[184] = 0.79388;
-   aVec[185] = 0.785348;
-   aVec[186] = 0.777295;
-   aVec[187] = 0.767854;
-   aVec[188] = 0.759619;
-   aVec[189] = 0.752358;
-   aVec[190] = 0.743394;
-   aVec[191] = 0.736674;
-   aVec[192] = 0.726982;
-   aVec[193] = 0.719321;
-   aVec[194] = 0.708937;
-   aVec[195] = 0.700003;
-   aVec[196] = 0.693202;
-   aVec[197] = 0.685961;
-   aVec[198] = 0.676889;
-   aVec[199] = 0.670513;
-   aVec[200] = 0.658788;
-   aVec[201] = 0.648343;
-   aVec[202] = 0.639381;
-   aVec[203] = 0.632761;
-   aVec[204] = 0.623614;
-   aVec[205] = 0.616164;
-   aVec[206] = 0.607884;
-   aVec[207] = 0.601437;
-   aVec[208] = 0.593944;
-   aVec[209] = 0.585817;
-   aVec[210] = 0.577248;
-   aVec[211] = 0.570328;
-   aVec[212] = 0.56147;
-   aVec[213] = 0.553509;
-   aVec[214] = 0.548765;
-   aVec[215] = 0.539964;
-   aVec[216] = 0.53343;
-   aVec[217] = 0.525787;
-   aVec[218] = 0.519868;
-   aVec[219] = 0.510817;
-   aVec[220] = 0.503274;
-   aVec[221] = 0.498478;
-   aVec[222] = 0.493058;
-   aVec[223] = 0.484176;
-   aVec[224] = 0.479896;
-   aVec[225] = 0.469998;
-   aVec[226] = 0.462893;
-   aVec[227] = 0.45576;
-   aVec[228] = 0.450375;
-   aVec[229] = 0.442683;
-   aVec[230] = 0.437887;
-   aVec[231] = 0.430736;
-   aVec[232] = 0.427216;
-   aVec[233] = 0.420775;
-   aVec[234] = 0.41485;
-   aVec[235] = 0.407769;
-   aVec[236] = 0.403546;
-   aVec[237] = 0.397028;
-   aVec[238] = 0.389534;
-   aVec[239] = 0.388569;
-   aVec[240] = 0.380296;
-   aVec[241] = 0.375312;
-   aVec[242] = 0.369457;
-   aVec[243] = 0.365884;
-   aVec[244] = 0.358889;
-   aVec[245] = 0.353344;
-   aVec[246] = 0.350364;
-   aVec[247] = 0.347136;
-   aVec[248] = 0.33872;
-   aVec[249] = 0.336706;
+	aVec = esps->endcap_shape; 
+	time_interval = esps->time_interval;
+	m_thresh = esps->endcap_thresh;
+    } else
+    {
+        time_interval = 1.0; 
+   	aVec.push_back(9.09091e-05);
```

Comment:
fill vector all at once

Reply:
I am not sure I understand the requested vector fill, could you please provide an example in C++ ?

Reply:
ok clarified by David, I believe the new codes uses the suggestion